MoreLinq/GroupAdjacent.cs

```diff
@@ -87,7 +87,7 @@ static partial class MoreEnumerable
             if (source == null) throw new ArgumentNullException(nameof(source));
             if (keySelector == null) throw new ArgumentNullException(nameof(keySelector));
 
-            return GroupAdjacent(source, keySelector, e => e, comparer);
+            return GroupAdjacent(source, keySelector, Identity<TSource>, comparer);
```

Comment:
Type argument is redundant:

```suggestion
            return GroupAdjacent(source, keySelector, Identity, comparer);
```